src/include/txn_inline.h

```diff
@@ -1377,6 +1379,20 @@ __wt_txn_modify_check(
     if (rollback) {
         WT_STAT_CONN_DATA_INCR(session, txn_update_conflict);
         ret = __wt_txn_rollback_required(session, "conflict between concurrent operations");
+
+        /* Dump information about the txn snapshot. */
+        if (WT_VERBOSE_LEVEL_ISSET(session, WT_VERB_TRANSACTION, WT_VERBOSE_DEBUG)) {
```

Comment:
Should we check if `ret` is `0` here? I am afraid that `ret` could be set on line 1381 to a different value and overridden by the different `WT_ERR` calls below.

Reply:
Should we find a way to:
- avoid setting `ret` lines 1386, 1387, 1389 and 1392?
- If one of those calls fail, go to the `err` label

Reply:
How about moving this code before calling `__wt_txn_rollback_required`? So that you don't have to worry about the errors? If they fail to allocate memory, let it return?

Reply:
Sounds good to me! See ab4afb021dec303bb89932be26c8671265e1d25e